nav2_simple_commander/nav2_simple_commander/line_iterator.py

```diff
@@ -0,0 +1,196 @@
+            self.m_ = (y1-y0)/(x1-x0)
+            self.b_ = y1 - (self.m_*x1)
+            if (self.b_ < 0):
+                self.equation_ = "y = " + str(self.m_) + "*x " + str(self.b_)
+            else:
+                self.equation_ = "y = " + str(self.m_) + "*x + " + str(self.b_)
+        elif x1 == x0 and y1 != y0:
+            self.valid_ = True
+            self.equation_ = "x = " + str(x1)
+        elif y1 == y1 and x1 != x0:
+            self.valid_ = True
+            self.m_ = (y1-y0)/(x1-x0)
+            self.b_ = y1 - (self.m_*x1)
+            self.equation_ = "y = " + str(y1)
+        else:
+            self.valid_ = False
+            self.equation_ = "Invalid"
+            raise ValueError(
+                "Line has zero length (All 4 points have same coordinates)")
+
+    def isValid(self):
+        """Check if line is valid."""
+        return self.valid_
+
+    def advance(self):
+        """Advance to the next point in the line."""
+        if self.x1_ > self.x0_:
+            if self.x_ < self.x1_:
+                self.x_ = round(self.clamp(
+                    self.x_ + self.step_size_, self.x0_, self.x1_), 5)
+                self.y_ = round(self.m_ * self.x_ + self.b_, 5)
+            else:
+                self.valid_ = False
+        elif self.x1_ < self.x0_:
+            if self.x_ > self.x1_:
+                self.x_ = round(self.clamp(
+                    self.x_ - self.step_size_, self.x1_, self.x0_), 5)
+                self.y_ = round(self.m_ * self.x_ + self.b_, 5)
+            else:
+                self.valid_ = False
+        else:
+            if self.y1_ > self.y0_:
+                if self.y_ < self.y1_:
+                    self.y_ = round(self.clamp(
+                        self.y_ + self.step_size_, self.y0_, self.y1_), 5)
+                else:
+                    self.valid_ = False
+            elif self.y1_ < self.y0_:
+                if self.y_ > self.y1_:
+                    self.y_ = round(self.clamp(
+                        self.y_ - self.step_size_, self.y1_, self.y0_), 5)
+                else:
+                    self.valid_ = False
+            else:
+                self.valid_ = False
+
+    def getX(self):
+        """Get the abscissa of the current point."""
+        return self.x_
+
+    def getY(self):
+        """Get the ordinate of the current point."""
+        return self.y_
+
+    def getX0(self):
+        """Get the abscissa of the initial point."""
+        return self.x0_
+
+    def getY0(self):
+        """Get the ordinate of the intial point."""
+        return self.y0_
+
+    def getX1(self):
+        """Get the abscissa of the final point."""
+        return self.x1_
+
+    def getY1(self):
+        """Get the ordinate of the final point."""
+        return self.y1_
+
+    def get_line_length(self):
+        """Get the length of the line."""
+        return sqrt(pow(self.x1_ - self.x0_, 2) + pow(self.y1_ - self.y0_, 2))
+
+    def get_line_equation(self):
```

Comment:
Can this be removed? I'm just not sure when this would be useful in practice. Also removes some of the stuff to populate `self.equation_` (+ is missing unit tests)

Reply:
Method removed.